### PR TITLE
fix(docker): usage of Ubuntu base image with pyenv and nvm

### PR DESCRIPTION
**Pull request by Divlo**
### What type of change does this PR introduce?
- [x] Bugfix
- [ ] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Not Sure?

### Does this PR introduce breaking changes?
- [ ] Yes
- [x] No

### The problem

This PR fixes Docker installation by fixing the start of Python TCP server.
The problem with the Docker installation is that we used Alpine Linux and `apk` package manager to install python, using the `apk add py3-pip` command but that installed the latest Python stable version (currently: Python v3.10.6), not what we explicitly wanted/required in `bridges/python/Pipfile`, that means `pipenv` didn't start the TCP server as it didn't have the correct Python version.

### The suggested solution

Instead, to use Linux Alpine, we use the Ubuntu base image, and install Node.js with `nvm` and Python with `pyenv`, as we do in the CLI.
This has the advantage to be closer to the "classical" installation (without docker).

Concerning implementation details, I will comment on lines that probably needs explication. :smile:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,10 +51,10 @@
     "train": "babel-node scripts/train/run-train.js",
     "prepare-release": "babel-node scripts/release/prepare-release.js",
     "check": "babel-node scripts/run-check.js",
-    "docker:build": "docker build -t leonai/leon .",
-    "docker:run": "docker-compose up",
-    "docker:dev": "docker-compose --file=docker-compose.dev.yml up",
-    "docker:check": "docker run --rm --interactive leonai/leon npm run check"
+    "docker:build": "docker build -t leon-ai/leon .",
+    "docker:run": "docker compose up",
```
Comment (Divlo) [inline package.json:55]: `docker-compose` is now deprecated, and we should use `docker compose`.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-      - '${LEON_PORT:-1337}:${LEON_PORT:-1337}'
+    image: 'leon-ai/leon'
     environment:
       LEON_PORT: ${LEON_PORT:-1337}
+      LEON_PY_TCP_SERVER_HOST: ${LEON_PY_TCP_SERVER_HOST:-0.0.0.0}
+      LEON_PY_TCP_SERVER_PORT: ${LEON_PY_TCP_SERVER_PORT:-1342}
     stdin_open: true
     tty: true
     command: 'npm run start'
+    network_mode: 'host'
```
Comment (Divlo) [inline docker-compose.yml:14]: `network_mode: 'host'` allows us to expose all ports inside the container including `1342` and `1337` (default ports used).

File: docker-compose.yml [hunk 3]
```diff
@@ -1,14 +1,14 @@
-version: '3.0'
 services:
-  leon-ai:
-    container_name: 'leon-ai'
+  leon:
+    container_name: 'leon'
     build:
       context: './'
-    image: 'leonai/leon'
```
Comment (Divlo) [inline docker-compose.yml:7]: As the organization is called `leon-ai` everywhere, not `leonai`, I find it nicer to call the image `leon-ai`.
This change is not strictly needed for this fix, but I guess it is a nice change to do, feel free to disagree, I can revert that one.

File: scripts/commit-msg.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ const commitEditMsgFile = '.git/COMMIT_EDITMSG'
 if (fs.existsSync(commitEditMsgFile)) {
   try {
     const commitMessage = fs.readFileSync(commitEditMsgFile, 'utf8')
-    const regex = '(build|BREAKING|chore|ci|docs|feat|fix|perf|refactor|style|test)(\\((web app|server|hotword|skill\\/([\\w-]+)))?\\)?: .{1,50}' // eslint-disable-line no-useless-escape
+    const regex = '(build|BREAKING|chore|ci|docs|feat|fix|perf|refactor|style|test)(\\((web app|docker|server|hotword|skill\\/([\\w-]+)))?\\)?: .{1,50}' // eslint-disable-line no-useless-escape
```
Comment (Divlo) [inline scripts/commit-msg.js:16]: Not strictly related to the fix, but I think a "docker" scope for commits messages is nice to have, if you don't want to have this scope, I can revert this change.

File: scripts/setup/setup-python-packages.js [hunk 5]
```diff
@@ -56,13 +56,17 @@ export default () => new Promise(async (resolve, reject) => {
         await installPythonPackages()
       } else {
         const dotProjectPath = path.join(process.cwd(), 'bridges/python/.venv/.project')
-        const dotProjectMtime = fs.statSync(dotProjectPath).mtime
+        if (fs.existsSync(dotProjectPath)) {
```
Comment (Divlo) [inline scripts/setup/setup-python-packages.js:59]: It happens that `bridges/python/.venv/.project` doesn't exist, so it throws an error when executing `fs.statSync`.
If this file doesn't exist, we install Python packages.

**Review by louistiti — APPROVED** [review]

**Comment by louistiti** [comment]
This is golden! Merged.

**Pull request merged**